src/components/base/Input.tsx

```diff
@@ -1,28 +1,24 @@
-import { useState } from "react";
+import React, { forwardRef, InputHTMLAttributes, useState } from "react";
 import { EyeIcon, EyeSlashIcon } from "@heroicons/react/24/solid";
 
-interface IInputFieldProps {
+export interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
+  error?: string;
   errors?: string[];
-  label?: string;
-  name: string;
-  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  label?: React.ReactNode;
+  name?: string;
   placeholder?: string;
   type: string;
-  value: string;
+  value?: string;
 }
 
-export default function InputField({
-  errors,
-  label,
-  name,
-  onChange,
-  placeholder,
-  type,
-  value,
-}: IInputFieldProps): JSX.Element {
+function Input(
+  { error, errors, label, name, placeholder, type, ...props }: InputProps,
+  ref: React.ForwardedRef<HTMLInputElement>,
+) {
   const [inputType, setInputType] = useState<string>(type);
+
   const handleShowPassword = () =>
-    inputType === "password" ? setInputType("text") : setInputType("password");
+    setInputType(inputType === "password" ? "text" : "password");
```

Comment:
This logic could be done outside of the input component, so if you want another button to trigger another event, like "Clear", "Check availability", you could do it without changing the component. Just FYI.

Reply:
yeah I would normally have an Input with a generic prop to render something inside the Input and to the right, so you could implement this logic in a more flexible way. For what the big challenge is I think it's okay to have such specific implementation here considering there's not that many inputs